core/src/main/java/org/bitcoinj/wallet/BasicKeyChain.java

```diff
@@ -177,8 +177,10 @@ private void importKeyLocked(ECKey key) {
             if (!key.isWatching() && isWatching)
                 throw new IllegalArgumentException("Key is not watching but chain is");
         }
-        ECKey previousKey = pubkeyToKeys.put(ByteString.copyFrom(key.getPubKey()), key);
-        hashToKeys.put(ByteString.copyFrom(key.getPubKeyHash()), key);
+        byte [] pubKey = key.getPubKey();
+        byte [] pubKeyHash = key.getPubKeyHash();
+        ECKey previousKey = pubkeyToKeys.put(ByteBuffer.wrap(Arrays.copyOf(pubKey, pubKey.length)), key);
```

Comment:
Why do we need to create a copy here?

Reply:
We don't if we rely on `key.getPubKey()` and `key.getPubKeyHash()` returning computed values and not some cache. Good point.